packages/credit-card-integration/src/CreditCardPaymentMethodComponent.tsx

```diff
@@ -328,82 +112,159 @@ class CreditCardPaymentMethodComponent extends Component<
                 language,
             })
         );
-    }
-
-    private handleUseNewCard: () => void = () => {
-        this.setState({
-            isAddingNewCard: true,
-            selectedInstrumentId: undefined,
-        });
     };
 
-    private handleSelectInstrument: (id: string) => void = (id) => {
-        this.setState({
-            isAddingNewCard: false,
-            selectedInstrumentId: id,
-        });
+    const handleUseNewCard = (): void => {
+        setIsAddingNewCard(true);
+        setSelectedInstrumentId(undefined);
     };
 
-    private handleDeleteInstrument: (id: string) => void = (id) => {
-        const {
-            paymentForm: { setFieldValue },
-        } = this.props;
-        const { instruments } = this.getCreditCardPaymentMethodDerivedProps();
-        const { selectedInstrumentId } = this.state;
+    const handleSelectInstrument = (id: string): void => {
+        setIsAddingNewCard(false);
+        setSelectedInstrumentId(id);
+    };
 
+    const handleDeleteInstrument = (id: string): void => {
         if (instruments.length === 0) {
-            this.setState({
-                isAddingNewCard: true,
-                selectedInstrumentId: undefined,
-            });
-
+            setIsAddingNewCard(true);
+            setSelectedInstrumentId(undefined);
             setFieldValue('instrumentId', '');
         } else if (selectedInstrumentId === id) {
-            this.setState({
-                selectedInstrumentId: this.getDefaultInstrumentId(),
-            });
+            const defaultInstrumentId = getDefaultInstrumentId();
 
-            setFieldValue('instrumentId', this.getDefaultInstrumentId());
+            setSelectedInstrumentId(defaultInstrumentId);
+            setFieldValue('instrumentId', defaultInstrumentId);
         }
     };
 
-    private getCreditCardPaymentMethodDerivedProps(): CreditCardPaymentMethodDerivedProps {
-        const { checkoutService, checkoutState, isUsingMultiShipping = false, method } = this.props;
+    useEffect(() => {
```

Comment:
I’ve used three separate `useEffect()` hooks here, each handling a different existing concern componentDidMount, componentDidUpdate, and componentWillUnmount.

Merging them into a single `useEffect()` would blur the distinction between dependencies and lifecycle behaviors.

However, I’m happy to merge them if we think that’s prefered. ⏩️